pennylane_lightning/lightning_tensor/lightning_tensor.py

```diff
@@ -0,0 +1,343 @@
+QuantumTapeBatch = Sequence[QuantumTape]
+QuantumTape_or_Batch = Union[QuantumTape, QuantumTapeBatch]
+PostprocessingFn = Callable[[ResultBatch], Result_or_ResultBatch]
+
+
+_backends = frozenset({"quimb"})
+# The set of supported backends.
+
+_methods = frozenset({"mps"})
+# The set of supported methods.
+
+
+def accepted_backends(backend: str) -> bool:
+    """A function that determines whether or not a backend is supported by ``lightning.tensor``."""
+    return backend in _backends
+
+
+def accepted_methods(method: str) -> bool:
+    """A function that determines whether or not a method is supported by ``lightning.tensor``."""
+    return method in _methods
+
+
+@simulator_tracking
+@single_tape_support
+class LightningTensor(Device):
+    """PennyLane Lightning Tensor device.
+
+    A device to perform tensor network operations on a quantum circuit.
+
+    Args:
+        wires (int): The number of wires to initialize the device with.
+            Defaults to ``None`` if not specified.
+        backend (str): Supported backend. Currently, only ``quimb`` is supported.
+        method (str): Supported method. Currently, only ``mps`` is supported.
+        shots (int): How many times the circuit should be evaluated (or sampled) to estimate
+            the expectation values. Currently, it can only be ``None``, so that computation of
+            statistics like expectation values and variances is performed analytically.
+        c_dtype: Datatypes for the tensor representation. Must be one of
+            ``np.complex64`` or ``np.complex128``.
+        **kwargs: keyword arguments. The following options are currently supported:
+
+            ``max_bond_dim`` (int): Maximum bond dimension for the MPS simulator.
+                It corresponds to the number of Schmidt coefficients retained at the end of the SVD algorithm when applying gates. Default is ``None``.
+            ``cutoff`` (float): Truncation threshold for the Schmidt coefficients in a MPS simulator. Default is ``1e-16``.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+
+    # So far we just consider the options for MPS simulator
+    _device_options = (
+        "backend",
+        "c_dtype",
+        "cutoff",
+        "method",
+        "max_bond_dim",
+    )
+
+    _new_API = True
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        *,
+        wires=None,
+        backend="quimb",
+        method="mps",
+        shots=None,
+        c_dtype=np.complex128,
+        **kwargs,
+    ):
+
+        if not accepted_backends(backend):
+            raise ValueError(f"Unsupported backend: {backend}")
+
+        if not accepted_methods(method):
+            raise ValueError(f"Unsupported method: {method}")
+
+        if shots is not None:
+            raise ValueError("LightningTensor does not support finite shots.")
+
+        super().__init__(wires=wires, shots=shots)
+
+        self._num_wires = len(self.wires) if self.wires else 0
```

Comment:
What are the implications of a 0-wire device in this case? Will there be any impact to how/where things runs?

Reply:
Good point. Making the `wires` argument optional comes from the epic's requirements. I still haven't tested extensively the device with 0 wires, but in the next PR there will be some tests for such a case (eventually, we can discuss how to proceed)